### PR TITLE
Bug 1193051 - Properly unregister pinning the web setting observer r=benfrancis

### PR DESCRIPTION
**Pull request by lissyx**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2fe8cd7fe935aaf68c6864da69865179a81afa0c

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=110ea0f545c6f4869b355305f8acab37dd15cdee

File: apps/system/js/app_chrome.js [hunk 1]
```diff
@@ -513,13 +521,8 @@
...
       // Adding or removing the click listener, depending on
       // the 'Pinning the Web' setting enabled or disabled
-      SettingsListener.observe(PINNING_PREF, '', function(enabled) {
-        var targets = [this.siteIcon, this.closePin];
-        var method = enabled ? 'addEventListener' : 'removeEventListener';
-        targets.forEach(function(element) {
-          element[method]('click', this);
-        }.bind(this));
-      }.bind(this));
+      this._boundPinningObserver = this._pinningObserver.bind(this);
```
Comment (mikehenrty) [inline apps/system/js/app_chrome.js]: Safer would be to make sure we never recreate the bound function:
```
this._boundPinningObserver = this._boundPinningObserver ||
  this._pinningObserver.bind(this);
```

Comment (benfrancis) [inline apps/system/js/app_chrome.js]: +1

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=dc4f4cb3069638f8a7c204958dfbf7c4b7cc3fcd

**Pull request merged**